src/utils/input-validation.ts

```
@@ -0,0 +1,318 @@
+  
+  // Command injection patterns - be more specific to avoid false positives
+  /[;&|`]\s*(rm|del|format|curl|wget|nc|netcat|bash|sh|cmd|powershell)/gi,
+  /\$\(\s*[a-z]/gi, // Command substitution like $(command)
+  /`[^`]*`/g, // Backtick command execution
+  
+  // Path traversal (including URL encoded)
+  /\.\.[\/\\]/g,
+  /%2e%2e[\/\\%2f]/gi, // URL encoded ../
+  
+  // Control characters (including null bytes)
+  /[\x00-\x1f\x7f-\x9f]/g,
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)